### DISCUSSION TITLE
Domain Model Diagram

### DISCUSSION THREAD

**Discussion by SebastianOpriel** [eclipse-edc/Connector#3815] [Q&A] [answered]
Hi, I found an overview of the main objects inside EDC in one of my documents. I ask myself if there is a new version and if yes, where I can find it. Maybe @tmberthold @efiege you have also a clue?

![image](https://github.com/eclipse-edc/Connector/assets/22075788/f7a439aa-35e2-4eac-9da3-5800bb5e1931)

BR
Sebastian

**Comment by ghost** [comment, marked as answer]
It's now in the Readme of the tractus-x edc: https://github.com/eclipse-tractusx/tractusx-edc/blob/main/docs/usage/README.md

**Reply by SebastianOpriel** [reply]
awesome, thanks

**Reply by rafaelmag110** [reply]
I wonder if those docs shouldn't be here instead, in the upstream Connector repository. On a first glance, the content doesn't seem to be too Catena-X specific and would complement very well the docs here. Any specific reason against?

**Reply by ghost** [reply]
It was at some point in time [here](https://github.com/eclipse-edc/Connector/pull/761/files#diff-17838484d795bdb53d49a0746b61fc89bc725fb45f5a3bea7beb61b380b780d0) but got [removed](https://github.com/eclipse-edc/Connector/pull/3591/files#diff-658a744f23d2da6733c63d8c68f7dd21869ab7f93d382269de4260442d0137c5) from here.

**Reply by SebastianOpriel** [reply]
+1 for that approach, but I assume there was some reason to remove that @ndr-brt or?

**Reply by ndr-brt** [reply]
it is outdated, there's no point in keeping outdated docs, especially the very technical as this diagram.
we should either have something code generated that stays up-to-date, or write up something more hi-level ([the Developer's Handbook](https://eclipse-edc.github.io/docs/#/documentation/developer/handbook) aims to cover that)

**Reply by jimmarino** [reply]
I agree completely with @ndr-brt about upstream documentation. Either the documentation should be very high-level or code generated using `AnnotationProcessors`. We already have a framework upstream for doing that.
